controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -149,7 +151,7 @@ func NewEndpointsWatcher(k8sAPI *k8s.API, log *logging.Entry) *EndpointsWatcher
 		UpdateFunc: func(_, obj interface{}) { ew.addService(obj) },
 	})
 
-	if !pkgK8s.EndpointSliceAccess(k8sAPI.Client) {
+	if !pkgK8s.EndpointSliceAccess(k8sAPI.Client, ew.enableEndpointSlices) {
```

Comment:
if we check for access in the main method as I suggest below, we can remove all of these checks in this file and instead treat the `ew.enableEndpointSlices` as authoritative.

Reply:
Amazing suggestion, it reads much better and makes matters less complicated. It also simplifies writing tests; I have uncommented and split up the unit test cases for EndpointSlices since they are no longer coupled to `EndpointSliceAccess`.